tests/estimators/speech_recognition/test_pytorch_icefall.py

```
@@ -0,0 +1,230 @@
+            "exp_dir": Path("."),
+            "lang_dir": Path("frames"),
+            # encoder/decoder params
+            "vocab_size": 3,  # blank, yes, no
+            "blank_id": 0,
+            "embedding_dim": 32,
+            "hidden_dim": 16,
+            "num_decoder_layers": 4,
+            "epoch": 6,
+            "avg": 1,
+        }
+    )
+
+    vocab_size = 1
+    try:
+        with open(Path(params.lang_dir) / "lexicon_disambig.txt") as lexicon_file:
+            for line in lexicon_file:
+                if len(line.strip()) > 0:  # and '<UNK>' not in line and '<s>' not in line and '</s>' not in line:
+                    vocab_size += 1
+    except:
+        pass
+    params.vocab_size = vocab_size
+
+    return params
+
+def get_transducer_model(params: AttributeDict):
+    from transducer.decoder import Decoder
+    from transducer.conformer import Conformer
+    from transducer.joiner import Joiner
+    from transducer.model import Transducer
+
+    encoder = Conformer(
+        num_features=params.feature_dim,
+        output_dim=params.hidden_dim,
+    )
+    decoder = Decoder(
+        vocab_size=params.vocab_size,
+        embedding_dim=params.embedding_dim,
+        blank_id=params.blank_id,
+        num_layers=params.num_decoder_layers,
+        hidden_dim=params.hidden_dim,
+        embedding_dropout=0.4,
+        rnn_dropout=0.4,
+    )
+    joiner = Joiner(input_dim=params.hidden_dim, output_dim=params.vocab_size)
+    transducer = Transducer(encoder=encoder, decoder=decoder, joiner=joiner)
+
+    return transducer
+
+
+def get_word2id(params):
+    from pathlib import Path
+
+    word2id = {}
+
+    # 0 is blank
+    id = 1
+    try:
+        with open(Path(params.lang_dir) / "lexicon_disambig.txt") as lexicon_file:
+            for line in lexicon_file:
+                if len(line.strip()) > 0:
+                    word2id[line.split()[0]] = id
+                    id += 1
+    except:
+        pass
```

The exception is being swallowed. It should be handled hence the CodeQL warning.